pipeline/console/tags.py

```diff
@@ -0,0 +1,144 @@
+import argparse
+import re
+import sys
+from typing import List
+
+from tabulate import tabulate
+
+from pipeline import PipelineCloud
+from pipeline.schemas.pagination import Paginated
+from pipeline.schemas.pipeline import (
+    PipelineTagCreate,
+    PipelineTagGet,
+    PipelineTagPatch,
+)
+from pipeline.util.logging import _print
+
+tag_re_pattern = re.compile(r"^[0-9a-zA-Z\-\_]+:[0-9a-zA-Z\-\_]+$", re.IGNORECASE)
```

Comment:
I used a slightly more involved regex for the API service, but at least this wil catch obvious mistakes client side.